core/src/main/groovy/org/groovymc/modsdotgroovy/plugin/PluginRegistry.groovy

```diff
@@ -15,9 +17,9 @@ final class PluginRegistry {
         }
     })
 
-    PluginRegistry() {
+    PluginRegistry(@Nullable Platform platform) {
         // Load plugins
-        plugins.addAll(ServiceLoader.load(ModsDotGroovyPlugin).asCollection())
+        plugins.addAll(ServiceLoader.load(ModsDotGroovyPlugin).findAll { ModsDotGroovyPlugin plugin -> platform == null || plugin.platforms.contains(platform) }.asCollection())
```

Comment:
Use `===` for null comparisons to avoid the overhead of going through Groovy Truth